### DISCUSSION TITLE
Memory leak within secondary thread.

### DISCUSSION THREAD

**Discussion by Asanoci** [agronholm/apscheduler#661] [Q&A]
Hello,

First of all, thank you for your app, it is very helpful.
I have a problem related to memory management within scheduled jobs.

In my app, I have a scheduled job that downloads files and loads them into memory through variables.
The problem is that every time the job runs, the memory is not freed, and it stacks at every job iteration.

I tried using the garbage collector but it did not work.

When i run the same job within the principal thread, i do not have any problem.

I don't know if the problem is understandable, if it isn't, i would be more specific :)

Thanks for your help !

**Comment by agronholm** [comment]
If you can provide a minimum workable example that demonstrates this problem, please open a bug report. I don't have the bandwidth to go hunt for memory leaks otherwise.
